x/superfluid/keeper/stake_test.go

```diff
@@ -519,6 +519,113 @@ func (suite *KeeperTestSuite) TestRefreshIntermediaryDelegationAmounts() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestDeleteAllEmptyIntermediaryAccounts() {
+	unbondingDuration := suite.App.StakingKeeper.GetParams(suite.Ctx).UnbondingTime
+	testCases := []struct {
+		name             string
+		validatorStats   []stakingtypes.BondStatus
+		superDelegations []superfluidDelegation
+		// each boolean field should account for each superfluid delegation
+		undelegationStart           []bool
+		timePassedBeforeEndblock    time.Duration
+		expInterAccNumAfterDeletion int
+	}{
+		{
+			"sinlge intermediary account deletion upon single superfluid undelegation finish",
+			[]stakingtypes.BondStatus{stakingtypes.Bonded},
+			[]superfluidDelegation{{0, 0, 0, 1000000}},
+			[]bool{true},
+			unbondingDuration,
+			0,
```

Comment:
0 by default
```suggestion
```

Reply:
deciding to leave this as having this visible is better for test readability imo